### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by waywardmonkeys**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -40,29 +35,22 @@ jobs:
...
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@master
         with:
           toolchain: stable
           components: clippy
-          profile: minimal
-          override: true
 
       - name: restore cache
         uses: Swatinem/rust-cache@v1
```
Comment (DataTriny) [inline .github/workflows/ci.yml:47]: Could you also bump this one to `v2` while at it please?

File: .github/workflows/generate-headers.yml [hunk 2]
```diff
@@ -9,11 +9,9 @@ jobs:
     - uses: actions/checkout@v3
 
     - name: install nightly toolchain
-      uses: actions-rs/toolchain@v1
+      uses: dtolnay/rust-toolchain@master
```
Comment (DataTriny) [inline .github/workflows/generate-headers.yml:12]:
```suggestion
      uses: dtolnay/rust-toolchain@nightly
```

File: .github/workflows/generate-headers.yml [hunk 3]
```diff
@@ -9,11 +9,9 @@ jobs:
...
       with:
```
Comment (DataTriny) [inline .github/workflows/generate-headers.yml:13]: The following lines should then be removed.

**Review by DataTriny — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (waywardmonkeys) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/generate-headers.yml (hunk 3 above)
Comment (waywardmonkeys) [reply .github/workflows/generate-headers.yml]: Done.

**Review by DataTriny — APPROVED** [review]
Looks good to me, thank you!

**Comment by mwcampbell** [comment]
@waywardmonkeys I finally got around to approving GitHub Actions for you in this repo. Looks like your changes broke a workflow. I'll let you look into it. Thanks.

**Comment by waywardmonkeys** [comment]
@mwcampbell This breakage isn't from me, but is from https://github.com/rust-lang/rust/issues/113152

I will submit a separate PR that fixes that and then this can move ahead after that.

**Comment by waywardmonkeys** [comment]
Once #261 lands, I will rebase this forward and that error should go away.

**Comment by waywardmonkeys** [comment]
@mwcampbell I've rebased this forward and the tests all pass now.

**Comment by waywardmonkeys** [comment]
Just checking back in on this (and my other PR).

**Pull request merged**